File: library/src/main/java/moe/leekcake/live2dforjvm/framework/type/CubismFramework.java
package moe.leekcake.live2dforjvm.framework.type;

import moe.leekcake.live2dforjvm.framework.jni.CubismFrameworkJNI;
import moe.leekcake.live2dforjvm.framework.type.id.CubismIdManager;

public class CubismFramework {
    public static void startUp() {
        CubismFrameworkJNI.StartUp();
    }

    public static void cleanUp() {
        CubismFrameworkJNI.CleanUp();
    }

    public static boolean isStarted() {
        return CubismFrameworkJNI.IsStarted();
    }

    public static void initialize() {
        CubismFrameworkJNI.Initialize();
    }

    public static void dispose() {
        CubismFrameworkJNI.Dispose();
    }

    public static boolean isInitialized() {
        return CubismFrameworkJNI.IsInitialized();
    }

    public static CubismIdManager getIdManager() {
        return new CubismIdManager(CubismFrameworkJNI.GetIdManager());
    }
}

File: library/src/main/kotlin/moe/leekcake/live2dforjvm/framework/type/CubismFramework.kt
package moe.leekcake.live2dforjvm.framework.type

import moe.leekcake.live2dforjvm.framework.CubismFrameworkJNI

object CubismFramework {
    fun startUp() {
        CubismFrameworkJNI.CubismFramework_StartUp()
    }

    fun cleanUp() {
        CubismFrameworkJNI.CubismFramework_CleanUp()
    }

    fun isStarted(): Boolean = CubismFrameworkJNI.CubismFramework_IsStarted()

    fun initialize() {
        CubismFrameworkJNI.CubismFramework_Initialize()
    }

    fun dispose() {
        CubismFrameworkJNI.CubismFramework_Dispose()
    }

    fun isInitialized(): Boolean {
        return CubismFrameworkJNI.CubismFramework_IsInitialized()
    }

    fun getIdManager(): CubismIdManager {
        return CubismIdManager(CubismFrameworkJNI.CubismFramework_GetIdManager())
    }
}